src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1191,22 +1198,49 @@ export function registerTerminalActions() {
 						ContextKeyEqualsExpr.create('view', TERMINAL_VIEW_ID),
 						ContextKeyExpr.not('config.terminal.integrated.showTabs')
 					]),
-				}]
+				}],
+				description: {
+					description: 'workbench.action.terminal.split',
+					args: [{
+						name: 'profile',
+						schema: {
+							type: 'object'
+						}
+					}]
+				},
 			});
 		}
-		async run(accessor: ServicesAccessor) {
+		async run(accessor: ServicesAccessor, profile?: ITerminalProfile) {
 			const terminalService = accessor.get(ITerminalService);
-			await terminalService.doWithActiveInstance(async t => {
-				const cwd = await getCwdForSplit(terminalService.configHelper, t, accessor.get(IWorkspaceContextService).getWorkspace().folders, accessor.get(ICommandService));
-				if (cwd === undefined) {
-					return undefined;
-				}
-				terminalService.splitInstance(t, { cwd });
-				return terminalService.showPanel(true);
-			});
+			if (profile) {
+				await terminalService.doWithActiveInstance(async t => {
+					const cwd = await getCwdForSplit(terminalService.configHelper, t, accessor.get(IWorkspaceContextService).getWorkspace().folders, accessor.get(ICommandService));
+					if (cwd === undefined) {
+						return undefined;
+					}
+					terminalService.splitInstance(t, profile);
```

Comment:
I think these branches can be merged and use:

```
terminalService.splitInstance(t, { profile, cwd });
```

Also I don't think right now the profile would respect the cwd which the above would fix.

Reply:
to do this, I'd need to convert the profile to a shellLaunchConfig or add an optional cwd to the profile

Reply:
had to make `convertProfileToShellLaunchConfig` public in terminal service bc can't pass both a `TerminalProfile `and `ShellLaunchConfig `to `splitInstance`.

Reply:
https://github.com/microsoft/vscode/blob/d3245fbf0247fcf35f07acd9ebc3c66fca915cd1/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L1213-L1225

https://github.com/microsoft/vscode/blob/d3245fbf0247fcf35f07acd9ebc3c66fca915cd1/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L934-L946

Reply:
Hmm, I'd prefer if `convertProfileToShellLaunchConfig` wasn't exposed, how about adding cwd to the `splitInstance` interface instead to minimize API changes?

```ts
	public splitInstance(instanceToSplit: ITerminalInstance, shellLaunchConfig?: IShellLaunchConfig): ITerminalInstance | null;
	public splitInstance(instanceToSplit: ITerminalInstance, profile: ITerminalProfile, cwd?: string | Uri): ITerminalInstance | null
	public splitInstance(instanceToSplit: ITerminalInstance, shellLaunchConfigOrProfile: IShellLaunchConfig | ITerminalProfile = {}, cwd?: string | Uri): ITerminalInstance | null {
```